src/lightning_app/cli/cmd_clusters.py

```diff
@@ -0,0 +1,238 @@
+MAX_CLUSTER_WAIT_TIME = 5400
+
+
+class AWSClusterManager:
+    """AWSClusterManager implements API calls specific to Lightning AI BYOC compute clusters when the AWS provider
+    is selected as compute backend."""
+
+    def __init__(self):
+        self.api_client = LightningClient()
+
+    def create(
+        self,
+        cost_savings: bool = False,
+        cluster_name: str = None,
+        role_arn: str = None,
+        region: str = "us-east-1",
+        external_id: str = None,
+        instance_types: [str] = [],
+        edit_before_creation: bool = False,
+        wait: bool = False,
+    ):
+        """request Lightning AI BYOC compute cluster creation.
+
+        :param cost_savings: flag indicating if the cluster should utilize cost savings mode
+        :param cluster_name: the name of the cluster to be created
+        :param role_arn: AWS IAM Role ARN used to provision resources
+        :param region: AWS region containing compute resources
+        :param external_id: AWS IAM Role external ID
+        :param instance_types: AWS instance types supported by the cluster
+        :param edit_before_creation: enable interactive editing of request before submitting it to Lightning AI
+        :param wait: wait for cluster to reach RUNNING state. use for debugging only
+        :return:
+        """
+        performance_profile = V1ClusterPerformanceProfile.DEFAULT
+        if cost_savings:
+            """In cost saving mode the number of compute nodes is reduced to one, reducing the cost for clusters
+            with low utilization."""
+            performance_profile = V1ClusterPerformanceProfile.COST_SAVING
+
+        body = V1CreateClusterRequest(
+            name=cluster_name,
+            spec=V1ClusterSpec(
+                cluster_type=V1ClusterType.BYOC,
+                performance_profile=performance_profile,
+                driver=V1ClusterDriver(
+                    kubernetes=V1KubernetesClusterDriver(
+                        aws=V1AWSClusterDriverSpec(
+                            region=region,
+                            role_arn=role_arn,
+                            external_id=external_id,
+                            instance_types=[V1InstanceSpec(name=x) for x in instance_types],
+                        )
+                    )
+                ),
+            ),
+        )
+        new_body = body
+        if edit_before_creation:
+            after = click.edit(json.dumps(body.to_dict(), indent=4))
+            if after is not None:
+                new_body = create_openapi_object(string2dict(after), body)
+            if new_body == body:
+                click.echo("cluster unchanged")
+
+        resp = self.api_client.cluster_service_create_cluster(body=new_body)
+        if wait:
+            _wait_for_cluster_state(self.api_client, resp.id, V1ClusterState.RUNNING)
+
+        click.echo(f"""${resp.id} cluster is ${resp.status.phase}""")
+
+    def list(self):
```

Comment:
let's keep some simple typing... :rabbit:

Reply:
the reference is for `mypy` - per the [PR check](https://github.com/Lightning-AI/lightning/runs/7566716618?check_suite_focus=true) no issues were found 👍🏼